server.R

```diff
@@ -50,17 +89,33 @@ server <- function(input, output,session) {
     if(input$subtype == "Multiplot" && length(input$genes) > 0) {
       query <- geneQuery(genes = input$genes, table = input$dataset)
       tcga <- dbGetQuery(database,query)
-
       query <- clinicalQuery(factors = "*", dataset = input$dataset)
       clinical <- dbGetQuery(database,query)
       clinical <- merge(tcga, clinical, by="UPN")
 
-      g <- ggplot(clinical,aes(fill=Gene, y=Expression, x=UPN, text = paste0("UPN ID: ",UPN,"<br />Dataset: ", input$dataset))) + geom_bar(position="dodge", stat="identity") + theme_bw() +
-        theme(text=element_text(size=12, family="avenir", face="bold"), axis.text=element_text(size=10, family="avenir", face="bold"),
-              axis.title=element_text(size=12, family="avenir", face="bold"),
-              axis.text.x = element_text(angle = 90, hjust = 1)) +
+      # Conditionally define the 'y' aesthetic and y-axis label
+      if (show.log2()) {
+        y_aes <- aes(y = Expression)
+        y_label <- "Log2 Expression"
+      } else {
+        y_aes <- aes(y = 2^Expression)
+        y_label <- "Raw Expression"
```

Comment:
One more minor change - instead of "Raw Expression", just use "Expression".  (there are several steps before it gets to this plotting point that alter the values, normalize them, etc)